test/functional/pdo_sqlsrv/pdostatement_fetchmode_emulate_prepare.phpt

```diff
@@ -1,167 +1,193 @@
 --TEST--
-PDO Fetch Mode Test with emulate prepare 
+PDO Fetch Mode Test with emulate prepare
 --DESCRIPTION--
-Basic verification for "PDOStatement::setFetchMode()�.
+Basic verification for "PDOStatement::setFetchMode()�.
 --ENV--
 PHPT_EXEC=true
 --SKIPIF--
-<?php require('skipif.inc'); ?>
+<?php require_once('skipif_mid-refactor.inc'); ?>
 --FILE--
 <?php
-include 'MsCommon.inc';
+require_once("MsCommon_mid-refactor.inc");
 
-function FetchMode()
-{
-	include 'MsSetup.inc';
+try {
+    $conn1 = connect();
 
-	$testName = "PDO Statement - Set Fetch Mode";
-	StartTest($testName);
-	
-	$dsn = "sqlsrv:Server=$server ; Database = $databaseName";
-	$conn1 = new PDO($dsn, $uid, $pwd);
-    $conn1->setAttribute( PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION );
+    // Prepare test table
+    $tableName = "pdo_test_table";
+    createTable($conn1, $tableName, array(new ColumnMeta("int", "ID", "NOT NULL PRIMARY KEY"), "Policy" => "varchar(2)", "Label" => "varchar(10)", "Budget" => "money"));
 
-	// Prepare test table
-	CreateTableEx($conn1, $tableName, "ID int NOT NULL PRIMARY KEY, Policy VARCHAR(2), Label VARCHAR(10), Budget MONEY", null);
+    try {
+        $res = $conn1->setAttribute(PDO::ATTR_EMULATE_PREPARES, true);
+        if ($res) {
+            echo "setAttribute should have failed.\n\n";
+        }
```

Comment:
No checking if connected with AE here? Because it should have failed anyway?

Reply:
yes